src/dia/attributes/calc.mjs

```diff
@@ -0,0 +1,55 @@
+const props = {
+    width: 'w',
+    height: 'h',
+    minimum: 's',
+    maximum: 'l',
+    diagonal: 'd'
+};
+const propsList = Object.keys(props).map(key => props[key]).join('');
+const numberPattern = '[-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?';
+const findSpacesRegex = /\s/g;
+const findExpressionsRegExp = /calc\(([^)]*)\)/g;
+const parseExpressionRegExp = new RegExp(`^(${numberPattern}\\*)?([${propsList}])([-+]${numberPattern})?$`, 'g');
+
+export function evalCalcExpression(expression, bbox) {
+    const match = parseExpressionRegExp.exec(expression.replace(findSpacesRegex, ''));
```

Comment:
I'm not so good at regex, but here you are just checking if the user supplied calc() expression matches the pattern in the RegExp object?

Reply:
Yes and read the values form the expression (multiplication factor, variable and addition/subtraction.